cli/azd/pkg/auth/manager.go

```diff
@@ -53,6 +53,31 @@ const useAzCliAuthKey = "auth.useAzCliAuth"
 // auth related configuration information (e.g. the home account id of the current user). This information is not secret.
 const authConfigFileName = "auth.json"
 
+// AzurePipelinesClientIDEnvVarName is the name of the environment variable that contains the client ID for the principal
+// to use when authenticating with Azure Pipelines via OIDC. It is set by both the AzureCLI@2 and AzurePowerShell@5 tasks
```

Comment:
Of note but not blocking: We encountered issues where these values are not set in `AzurePowerShell@5` in Windows https://github.com/Azure/azure-sdk-for-net/pull/44684 

This can be mitigated by following the steps in that PR. No change needed here.